crazyflie_sim/crazyflie_sim/visualization/blender.py

```diff
@@ -0,0 +1,208 @@
+import bpy
+import numpy as np
+import os
+import datetime
+import rowan as rw
+import yaml
+from rclpy.node import Node
+from ..sim_data_types import State, Action
+
+# rotation vectors are axis-angle format in "compact form", where
+# theta = norm(rvec) and axis = rvec / theta
+# they can be converted to a matrix using cv2. Rodrigues, see
+# https://docs.opencv.org/4.7.0/d9/d0c/group__calib3d.html#ga61585db663d9da06b68e70cfbf6a1eac
+def opencv2quat(rvec):
+    angle = np.linalg.norm(rvec)
+    if angle == 0:
+        q = np.array([1,0,0,0])
+    else:
+        axis = rvec.flatten() / angle
+        q = rw.from_axis_angle(axis, angle)
+    return q
+
+class Visualization:
+
+    def __init__(self, node: Node, params: dict, names: list[str], states: list[State]):
+        self.node = node
+
+        ## blender 
+        # load environment
+        world = bpy.context.scene.world
+        world.use_nodes = True
+        background_paths = [""]
+        self.env = world.node_tree.nodes.new("ShaderNodeTexEnvironment")
+        for subdir, dirs, files in os.walk("src/crazyswarm2/crazyflie_sim/data/env"):
```

Comment:
This will only work, if the ros2 launch command is executed directly from the workspace. To make it agnostic, you could use `pathlib` and `__file__`, or use the ROS2-specific packages to query for the folder with `get_package_share_directory`. (three more places like that below)